src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -1130,30 +1146,29 @@ RESULT_CODE ProtocolHandlerImpl::HandleControlMessageStartSession(
         }
         return RESULT_OK;
       }
+      // This is non navi application,
+      // so that a PTU must be triggered if no certificate is provided
+      // and PTU has not been triggered by this reason
       sync_primitives::AutoLock lock(ptu_state_lock_);
+      // PTU has not been triggered
+      // trigger PTU
       if (ptu_state_ == kNotTriggered) {
         ptu_state_ = kTriggered;
-        pending_session_ = SessionInfo(connection_id,
-                                       packet.session_id(),
-                                       protocol_version,
-                                       packet.service_type(),
-                                       si.hash_id_,
-                                       si.service_exists_);
+        pending_session_ = session_info;
         security_manager_->NotifyOnCertififcateUpdateRequired();
       } else if (ptu_state_ != kTriggered) {
+        // PTU has already been triggered
+        // and had brought no valid certificate
+        // Nack should be sent
         SendStartSessionNAck(connection_id,
                              packet.session_id(),
                              protocol_version,
                              packet.service_type());
       }
       return RESULT_OK;
     }
-    struct SessionInfo session_info = SessionInfo(connection_id,
-                                                  packet.session_id(),
-                                                  protocol_version,
-                                                  packet.service_type(),
-                                                  si.hash_id_,
-                                                  si.service_exists_);
+    // Certificate is existing and doesn't rquire update,
+    // try to start encrypted service
     StartEncryptedService(session_info);
```

Comment:
@tkireva Please be mentioned that in OnSessionStartedCallback service has been already started